src/databricks/labs/ucx/recon/_base.py

```diff
@@ -0,0 +1,78 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+
+
+@dataclass
+class TableDescriptor:
```

Comment:
we already have a Table dataclass that you can reuse
https://github.com/databrickslabs/ucx/blob/953c4837614b355d03bac9c0f201ca2a39e11a14/src/databricks/labs/ucx/hive_metastore/tables.py#L50-L54